java-manta-client/src/main/java/com/joyent/manta/client/RecursiveDirectoryCreationStrategy.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright (c) 2017, Joyent, Inc. All rights reserved.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.joyent.manta.client;
+
+import com.joyent.manta.exception.MantaClientHttpResponseException;
+import com.joyent.manta.exception.MantaErrorCode;
+import com.joyent.manta.http.MantaHttpHeaders;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * Base class for classes which implement recursive directory creation.
+ *
+ * @author <a href="https://github.com/tjcelaya">Tomas Celayac</a>
+ * @since 3.1.7
+ */
+abstract class RecursiveDirectoryCreationStrategy {
+
+    @SuppressWarnings("checkstyle:JavaDocVariable")
```

Comment:
Let's see if we can globally suppress this check in our checkstyle settings because this is just silly.

Reply:
```<module name="JavadocVariable">
    <property name="ignoreNamePattern" value="LOG|LOGGER"/>
</module>```

added to `checkstyle.xml`